typescript/solver/config/types.ts

```diff
@@ -0,0 +1,41 @@
+import z from "zod";
+import { ethers } from "ethers";
+
+export const addressSchema = z
+  .string()
+  .refine((address) => ethers.utils.isAddress(address), {
+    message: "Invalid address",
+  });
+export const addressValueSchema = z.union([
+  addressSchema,
+  z.array(addressSchema).default([]),
+  z.literal("*"),
+]);
+
+export const domainSchema = z
```

Comment:
IMO, it would be nice to use the domain name strings in addition to/instead of the domain numbers